server/controllers/finance/reports/cashflow/index.js

```diff
@@ -553,11 +551,16 @@ function document(req, res, next) {
     if (incomes) {
       incomes.forEach((item) => {
         tempIncome[item.origin_id] = typeof tempIncome[item.origin_id] !== 'undefined';
+        // FIX ME
+        // typeof tempIncome[item.origin_id] !== 'undefined' It's TRUE
+        // But this affectation " tempIncome[item.origin_id] = typeof tempIncome[item.origin_id] !== 'undefined' " is FALSE ???
+        // if (tempIncome[item.origin_id] === true) {  >> Not tested TRUE
 
-        if (tempIncome[item.origin_id] === true) {
+        if (typeof tempIncome[item.origin_id] !== 'undefined') {
```

Comment:
Why don't you store this in a temporary variable?

```js
var hasIncomeValue= typeof tempIncome[item.origin_id] !== 'undefined';
tempIncome[item.origin_id] = hasIncomeValue;
if (hasIncomeValue) {
  // do something
}
```

If the `undefined` check is giving you problems, you can try `_.isUndefined(tempIncome[item.origin_id])`.  [Here is the documentation](https://lodash.com/docs/4.17.4#isUndefined).